tools/api-coverage/src/api.rs

```diff
@@ -0,0 +1,51 @@
+use std::collections::HashSet;
+
+use anyhow::Result;
+use syn::visit::{self, Visit};
+
+use crate::util;
+
+/// Used by APIVisitor to get the function name being called.
+#[derive(Default)]
+struct FnNameVisitor {
+    segments: Vec<String>,
+}
+
+impl<'ast> Visit<'ast> for FnNameVisitor {
+    fn visit_path_segment(&mut self, node: &'ast syn::PathSegment) {
+        let ident = format!("{}", node.ident);
+        self.segments.push(ident);
+    }
+}
+
+/// A visitor to collect all function calls in the API crate.
+#[derive(Default)]
+struct APIVisitor {
+    calls: HashSet<String>,
+}
+
+impl<'ast> Visit<'ast> for APIVisitor {
+    fn visit_expr_call(&mut self, node: &'ast syn::ExprCall) {
+        if let syn::Expr::Path(path) = node.func.as_ref() {
+            let mut fnname = FnNameVisitor::default();
+            visit::visit_expr_path(&mut fnname, path);
+            let fnname = fnname.segments.join("::");
+            self.calls.insert(fnname);
+        } else {
+            println!("Weird guy here!: {:?}", node.func);
```

Comment:
What does this case mean? You expect it to be unreachable?

Reply:
That was just my in-progress learning of the AST. I'll make it a panic now that I've shown we don't have any thing in our code base that doesn't have a PathExpr for the function invocation expression.

Reply:
Fixed.